### ISSUE TITLE
CacheListenerTest may fail with NPE when running against valid implementation

### ISSUE THREAD

**Issue by vbekiaris** [jsr107/jsr107tck#140]
[Javadoc of `CacheEntryEvent#getValue`](https://github.com/jsr107/jsr107spec/blob/master/src/main/java/javax/cache/event/CacheEntryEvent.java#L60) states:

```
   * Returns the same value as {@link #getOldValue()} for
   * {@link CacheEntryExpiredListener} and {@link CacheEntryRemovedListener}.
```

[Javadoc of `CacheEntryEvent#getOldValue`](https://github.com/jsr107/jsr107spec/blob/master/src/main/java/javax/cache/event/CacheEntryEvent.java#L77-L83) states:

```
   * The old value may be available for {@link CacheEntryUpdatedListener},
   * {@link CacheEntryExpiredListener} and {@link CacheEntryRemovedListener}
   * if {@link CacheEntryListenerConfiguration#isOldValueRequired()} is false.
```

Given the above, a valid implementation may choose to not include the removed / expired value when notifying `CacheEntryExpiredListener`/`CacheEntryRemovedListener` registered with `isOldValueRequired==false`. In this case, both `CacheEntryEvent#getValue` and `CacheEntryEvent#getOldValue` will return `null`. However the entry event filter used in `CacheListenerTest#testFilteredListener` assumes `CacheEntryEvent#getValue()` is never null:

```
private static class MyCacheEntryEventFilter implements CacheEntryEventFilter<Long, String>, Serializable {
      @Override
      public boolean evaluate(
          CacheEntryEvent<? extends Long, ? extends String> event)
          throws CacheEntryListenerException {
          return event.getValue().contains("a") ||
              event.getValue().contains("e") ||
              event.getValue().contains("i") ||
              event.getValue().contains("o") ||
              event.getValue().contains("u");
      }
  }
```

If my understanding is correct, then this looks like a defect in `MyCacheEntryEventFilter` that should be fixed to take into account the possibility of `event.getValue()` returning `null`.

(TCK version 1.1.0)

**Comment by cruftex** [comment]
See tables here: https://github.com/jsr107/jsr107spec/issues/391#issuecomment-339224104

> If my understanding is correct, then this looks like a defect in `MyCacheEntryEventFilter` that should be fixed to take into account the possibility of `event.getValue()` returning `null`.

I think the same. Tests should be fixed.

These ambiguous things are a problem because the RI is only doing it "one way". We should have the "different" semantics somehow in the RI. At least mention it. Ideally with a "switch" to test that the TCK is doing the right thing in every case.

**Comment by vbekiaris** [comment]
@cruftex thanks for the quick feedback, I created a fix PR here: #142
